src/main/java/org/saltyrtc/client/cookie/CookiePair.java

```diff
@@ -54,13 +56,9 @@ public Cookie getOurs() {
         return this.ours;
     }
 
-    public boolean hasTheirs() {
-        return this.theirs != null;
-    }
-
     @Nullable
-    public Cookie getTheirs() {
-        return this.theirs;
+    public Optional<Cookie> getTheirs() {
```

Comment:
IIRC we cannot use `Optional` because older Android versions need to be supported.

Reply:
Yep, very unfortunate but true.

Reply:
What about TaskHelper?
https://github.com/saltyrtc/saltyrtc-client-java/blob/87fd191daae0f2ff727de497f218609d90facda4/src/main/java/org/saltyrtc/client/helpers/TaskHelper.java#L41

Reply:
Must have been an oversight. Unless this can now be desugared for API level < 24.